src/codegen/doc/mod.rs

```diff
@@ -179,7 +179,24 @@ fn generate_doc(w: &mut dyn Write, env: &Env) -> Result<()> {
                     .find(move |f| &f.glib_name == c_identifier)
                     .map(|f| f.doc_trait_name.as_ref())
                     .flatten();
-                let parent = doc_trait_name.map(|p| Box::new(TypeStruct::new(SType::Trait, p)));
+                let doc_struct_name = (&global_functions.functions)
+                    .iter()
+                    .find(move |f| &f.glib_name == c_identifier)
+                    .map(|f| f.doc_struct_name.as_ref())
+                    .flatten();
```

Comment:
```suggestion
                    .and_then(|f| f.doc_struct_name.as_ref());
```

?

Reply:
ah, for whatever reason i didn't notice we have the same thing a few lines above that uses and_then :)